extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+    const interpolate = numeric(fromPercentage, toPercentage);
+    const curve = bezierCurve(0.4, 0, 0.2, 1); // fast-out-slow-in
+    const duration = 200;
+    // Single animation ensure elements have props updated at same pace
+    // Multiple animations would be scheduled at different raf
+    return Animation.animate(this.element, pos => {
+      this.updatePositions(interpolate(pos));
+    }, duration, curve).thenAlways();
+  }
+
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  dragStart(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    this.win.addEventListener('mousemove', this.dragMove);
+    this.win.addEventListener('mouseup', this.dragEnd);
+
+    this.moveOffset_ = e.pageX;
+    this.splitOffset_ = this.bar_./*OK*/getBoundingClientRect().left;
+  }
+  /**
+   * Handle drag move
+   * @param {Event} e
+   */
+  dragMove(e) {
+    e.preventDefault();
+    e.stopPropagation(); // avoid clashing with clickImage
+    this.pointerMoveX_(e.pageX);
+  }
+  /**
+   * Remove listeners on drag end
+   * e is optional since dragEnd will also be used for unregister cleanup
+   * @param {Event|null} e
+   */
+  dragEnd(e) {
+    if (e) {
+      e.stopPropagation();
+    }
+
+    this.win.removeEventListener('mousemove', this.dragMove);
+    this.win.removeEventListener('mouseup', this.dragEnd);
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+  }
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  touchStart(e) {
+    this.container_.addEventListener('touchmove', this.touchMove);
+    this.container_.addEventListener('touchend', this.touchEnd);
+
+    this.moveOffset_ = e.touches[0].pageX;
+    this.splitOffset_ = this.bar_./*OK*/getBoundingClientRect().left;
+  }
+  /**
+   * Handle touch move
+   * @param {Event} e
+   */
+  touchMove(e) {
+    // When we are holding bar button (or its icon)
+    // Do not scroll page
+    if (e.target === this.barButton_ ||
+        (this.barButtonIcon_ && e.target === this.barButtonIcon_)) {
+      e.preventDefault();
+    }
+    e.stopPropagation(); // avoid clashing with clickImage
+    if (e.touches.length > 0) {
+      this.pointerMoveX_(e.touches[0].pageX);
+    }
+  }
+  /**
+   * Cleanup when touch released
+   * @param {Event|null} e
+   */
+  touchEnd(e) {
+    if (e) {
+      // Avoid bubbling up to element
+      e.stopPropagation();
+    }
+    this.container_.removeEventListener('touchmove', this.touchMove);
+    this.container_.removeEventListener('touchend', this.touchEnd);
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+  }
+
+  /**
+   * Pointer move X logic
+   * @param {number} pointerX
+   * @private
+   */
+  pointerMoveX_(pointerX) {
+    // const {width} = this.getLayoutBox();
+    const {width} = this.container_./*OK*/getBoundingClientRect();
+    const {left: leftBound, right: rightBound}
+    //    = this.getLayoutBox();
+        = this.container_./*OK*/getBoundingClientRect();
```

Comment:
No need to call `this.container_./*OK*/getBoundingClientRect();` twice here. ;)  
But per other comments, you should cache this and only update when necessary. =)